src/Cli/dotnet/ToolPackage/ToolPackageDownloader.cs

```diff
@@ -93,6 +93,12 @@ internal class ToolPackageDownloader : IToolPackageDownloader
                     var nugetPackageDownloader = new NuGetPackageDownloader.NuGetPackageDownloader(toolDownloadDir, verboseLogger: nugetLogger, isNuGetTool: true);
 
                     var packageSourceLocation = new PackageSourceLocation(packageLocation.NugetConfig, packageLocation.RootConfigDirectory, null, packageLocation.AdditionalFeeds);
+
+                    bool givenSpecificVersion = false;
+                    if (!string.IsNullOrEmpty(versionRange.OriginalString))
```

Comment:
I don't think the fact that it has an OriginalString necessarily means it is for a specific version.  The original string could be "*" or something.

I think it would be better to check that the min and max version are the same, similar to what you did in `GetBestPackageVersionAsync`:

```c#
if(versionRange.MinVersion != null && versionRange.MaxVersion != null && versionRange.MinVersion == versionRange.MaxVersion)
```